proto/redpanda/api/dataplane/v1alpha1/pipeline.proto

```diff
@@ -0,0 +1,212 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message Pipeline {
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.len = 20
+  ];
+
+  string display_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+
+  string description = 3 [(buf.validate.field).string.max_len = 1000];
+
+  // The configuration of the Pipeline as YAML.
+  // See https://docs.redpanda.com/redpanda-connect/configuration/about/
+  string config_yaml = 4 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The configuration of the Pipeline as YAML. See https://docs.redpanda.com/redpanda-connect/configuration/about for more details."}
+  ];
+
+  // The minimum amount of resources needed for the pipeline to run
+  Resources request = 5 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The minimum amount of resources needed for the pipeline to run."}
+  ];
+
+  // The maximum amount of resources that should be assigned to the pipeline
+  Resources limit = 6 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The maximum amount of resources that should be assigned to the pipeline."}];
+
+  Status status = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  message Spec {}
+
+  message Resources {
+    string memory_shares = 1;
+    string cpu_shares = 2;
+  }
+
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    STATE_STARTING = 1;
+    STATE_RUNNING = 2;
+    STATE_STOPPING = 3;
+    STATE_STOPPED = 4;
+    STATE_ERROR = 5;
+  }
+
+  message Status {
+    State state = 1;
+    string error = 2;
+  }
+}
+
+service PipelineService {
```

Comment:
Because my understanding is that we're following a [resource-oriented design](https://google.aip.dev/121) where each resource has its own service.

RedpandaConnectService would be too broad and doesn't refer to a single resource.

Instead if we want to manage more resources we should create extra services for each resource. For example `SecretService`.